api/pkg/s3/multipartuploadinit.go

```diff
@@ -16,133 +16,57 @@ package s3
-
-	var client datastore.DataStoreAdapter
-	if backendName != "" {
-		object.Backend = backendName
-		client = getBackendByName(ctx, s, backendName)
-	} else {
-		bucket, _ := s.s3Client.GetBucket(ctx, &s3.CommonRequest{Context: actx.ToJson(), Id: bucketName})
-		object.Backend = bucket.Backend
-		client = getBackendClient(ctx, s, bucketName)
-	}
-	if client == nil {
-		response.WriteError(http.StatusInternalServerError, NoSuchBackend.Error())
-		return
-	}
-	res, s3err := client.InitMultipartUpload(&object, ctx)
-	if s3err != NoError {
-		response.WriteError(http.StatusInternalServerError, s3err.Error())
+	if !isValidObjectName(objectKey) {
+		log.Errorf("object name is not valid.")
+		WriteErrorResponse(response, request, ErrInvalidObjectName)
 		return
 	}
 
-	lastModified := time.Now().Unix()
-	record := s3.MultipartUploadRecord{ObjectKey: objectKey, Bucket: bucketName, Backend: object.Backend, UploadId: res.UploadId}
-	record.InitTime = lastModified
-	record.TenantId = actx.TenantId
-	record.UserId = actx.UserId
-	_, err := s.s3Client.AddUploadRecord(context.Background(), &record)
+	acl, err := getAclFromHeader(request)
 	if err != nil {
-		client.AbortMultipartUpload(res, ctx)
-		response.WriteError(http.StatusInternalServerError, s3err.Error())
+		log.Errorf("failed to get acl from http header, err:", err)
+		WriteErrorResponse(response, request, err)
 		return
 	}
 
-	// Currently, only support tier1 as default
-	tier := int32(utils.Tier1)
-	object.Tier = tier
-	// standard as default
-	object.StorageClass = constants.StorageClassOpenSDSStandard
-
-	object.ObjectKey = objectKey
-	objectInput := s3.GetObjectInput{Bucket: bucketName, Key: objectKey}
-	objectMD, _ := s.s3Client.GetObject(ctx, &objectInput)
-	if objectMD != nil {
-		objectMD.ObjectKey = objectKey
-		objectMD.BucketName = bucketName
-		objectMD.InitFlag = "0"
-		objectMD.IsDeleteMarker = ""
-		objectMD.Partions = nil
-		objectMD.Backend = object.Backend
-		objectMD.Size = int64(size)
-		objectMD.LastModified = lastModified
-		objectMD.Tier = object.Tier
-		objectMD.StorageClass = object.StorageClass
-		//insert metadata
-		_, err := s.s3Client.CreateObject(ctx, objectMD)
-		if err != nil {
-			log.Errorf("err is %v\n", err)
-			response.WriteError(http.StatusInternalServerError, err)
-			client.AbortMultipartUpload(res, ctx)
-			return
-		}
-	} else {
-		object.Size = int64(size)
-		object.LastModified = lastModified
-		object.InitFlag = "0"
+	// Save metadata.
+	attr := extractMetadataFromHeader(request)
 
-		//insert metadata
-		_, err := s.s3Client.CreateObject(ctx, &object)
-		if err != nil {
-			log.Errorf("err is %v\n", err)
-			response.WriteError(http.StatusInternalServerError, err)
-			client.AbortMultipartUpload(res, ctx)
-			return
-		}
-	}
-
-	result := model.InitiateMultipartUploadResult{
-		Xmlns:    model.Xmlns,
-		Bucket:   res.Bucket,
-		Key:      res.Key,
-		UploadId: res.UploadId,
+	storageClass, err := getStorageClassFromHeader(request)
```

Comment:
This definition is not compatible with the one we defined in s3 service. Consider that we need to support different backends which may have different definitions for storage class, we had an map which is defined in s3/pkg/service/service.go, could you please use the definition in s3 service?

Reply:
@sfzeng , I will comment storage class except STANDARD, STANDARD_IA,GLACIER.  In the api layer, we only check if storage class header is one of the three storage class, and in the s3 layer,  I add a new issue to pass storage class to backend